### ISSUE TITLE
Add ollama support [ENHANCEMENT]

### ISSUE THREAD

**Issue by nonetrix** [mike-fabian/ibus-typing-booster#502]
**Is your feature request related to a problem? Please describe.**
I think it would be super neat if you could use LLMs to predict text, they are basically auto complete on steroids and take everything you typed before into account. Even a smaller LLM like Phi 2 would be great, which would be likely what I would use because it's very light and runs stupidly fast even on CPU and absurdly fast on GPU. I know AI is controversial, but I think auto complete is a great use case for it 

**Describe the solution you'd like**
A option to point it to a ollama daemon running on your computer and what model to use for text prediction, it should ideally take into account the last few worlds or tokens you have written

**Describe alternatives you've considered**
There is also OpenAI but that isn't locally on your computer and would cost a lot for the user, also you could probably use llama cpp instead which has Python bindings but this might make GPU acceleration harder. I would just go with ollama since you can get it to work with both AMD ROCm and NVIDIA CUDA, also it has a deamon and a nice API it's just all around nice

**Additional context**
[Ollama](https://github.com/ollama/ollama) Github

**Comment by mike-fabian** [comment]
Yes, I have also started to think about this already and this is certainly something I want to look into soon.

**Comment by mike-fabian** [comment]
https://github.com/mike-fabian/ibus-typing-booster/releases/tag/2.28.0 has support for doing an AI chat using ollama or ramalama.

Documentation: https://mike-fabian.github.io/ibus-typing-booster/docs/user/#ai-chat

This is some sort of “ollama support” although it does  something different than text prediction. 

I have started a bit investigating how to do text prediction with ollama.

Actually that seems to be a bit difficult at the moment as ollama does not support `logprobs`:

 https://github.com/ollama/ollama/issues/2415
https://github.com/ollama/ollama-python/issues/517

```
mfabian@hathi:~
$ curl http://localhost:11434/v1/completions -d '{
  "prompt": "The cat sat on the",
  "logprobs": 9,
  "max_tokens": 1
}'
{"error":{"message":"model '' not found","type":"api_error","param":null,"code":null}}
mfabian@hathi:~
$ curl http://localhost:11434/api/version
{"version":"0.9.6"}mfabian@hathi:~
$ curl http://localhost:11434/v1/completions -d '{
  "prompt": "The cat sat on the",
  "logprobs": 9,
  "max_tokens": 1
}'
{"error":{"message":"model '' not found","type":"api_error","param":null,"code":null}}
mfabian@hathi:~
$ 
```
ramalama does support it though and that looks promising:

```
$ ramalama serve --network=host -p 11434 hf://lmstudio-community/Mistral-7B-Instruct-v0.3-GGUF/Mistral-7B-Instruct-v0.3-Q4_K_M.gguf
```
And then in another terminal:
```
mfabian@hathi:~
$ curl http://localhost:11434/v1/completions -d '{
  "prompt": "The cat sat on the",
  "logprobs": 9,
  "max_tokens": 1
}'
{"choices":[{"text":" windows","index":0,"logprobs":{"content":[{"id":9180,"token":" windows","bytes":[32,119,105,110,100,111,119,115],"logprob":-1.9117858409881592,"top_logprobs":[{"id":9180,"token":" windows","bytes":[32,119,105,110,100,111,119,115],"logprob":-1.9117858409881592},{"id":2378,"token":" mat","bytes":[32,109,97,116],"logprob":-2.994535207748413},{"id":3169,"token":" table","bytes":[32,116,97,98,108,101],"logprob":-2.994577169418335},{"id":9913,"token":" roof","bytes":[32,114,111,111,102],"logprob":-3.0396344661712646},{"id":17978,"token":" fence","bytes":[32,102,101,110,99,101],"logprob":-3.165278196334839},{"id":3692,"token":" window","bytes":[32,119,105,110,100,111,119],"logprob":-3.3491570949554443},{"id":6126,"token":" edge","bytes":[32,101,100,103,101],"logprob":-3.3547096252441406},{"id":5134,"token":" floor","bytes":[32,102,108,111,111,114],"logprob":-3.578108549118042},{"id":4268,"token":" wall","bytes":[32,119,97,108,108],"logprob":-3.641547918319702}]}]},"finish_reason":"length"}],"created":1759261625,"model":"lmstudio-community/Mistral-7B-Instruct-v0.3-GGUF/Mistral-7B-Instruct-v0.3-Q4_K_M.gguf","system_fingerprint":"b1-b0d5299","object":"text_completion","usage":{"completion_tokens":1,"prompt_tokens":6,"total_tokens":7},"id":"chatcmpl-mYTN6JxuxiCUfvr3fYPLfKqkloYercln","timings":{"cache_n":0,"prompt_n":6,"prompt_ms":2511.867,"prompt_per_token_ms":418.64450000000005,"prompt_per_second":2.38866150158428,"predicted_n":1,"predicted_ms":1.681,"predicted_per_token_ms":1.681,"predicted_per_second":594.883997620464}}mfabian@hathi:~
$ 
````

> To my understanding LLMs technically give you multiple likely tokens with different weights

The LLMs can do that, but at the moment this feature is not accessible via ollama. 

With ollama, at the moment one can only try to workaround this by asking a chat question like:

```
Please give me the 9 most
plausible single words which  could 
follow after "The cat sat on the",
 sorted by plausability. Answer with
 a sorted list of strings in Python
 list format please. Answer **only** with
 that list, nothing else so I can use 
 `eval(your_answer)` to get the list.
```
But the response to such a question is slow and it is hard to phrase the question in a way that one really gets only the list. Sometimes one gets other garbage or somewhat different formats and then parsing the list out of the result might fail.

Here I try this with various models using the recently added ollama support in Typing Booster:

https://github.com/user-attachments/assets/b4978586-9890-48a5-a8cc-88392a3293ee

With smaller models like `phi2` the results are worse and still too slow. 

If `logprob` is supported, one get such results **much** faster and in an easily parsable json format.

As ollama does not support logprobs at the moment, I’ll try whether I can make this work with ramalama  first. 

https://github.com/codelion/optillm might be an alternative to ramalama, it also seems to support logprobs already.
